src/main/java/com/teragrep/lsh_01/pool/RelpConnectionPool.java

```diff
@@ -63,6 +73,11 @@ public IRelpConnection take() {
             if (frameDelegate == null) {
                 frameDelegate = relpConnectionWrapSupplier.get();
             }
+            else if (frameDelegate.recordsSent() >= rebindRequestAmount && rebindEnabled) {
```

Comment:
this works only if the connection is taken and put into pool between records sent, I wonder if it should be within the ManagedConnection